components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -2154,13 +2162,36 @@ private static void updateCookieConfig(CookieBuilder cookieBuilder, IdentityCook
     public static String getMultiAttributeSeparator() {
 
         String multiAttributeSeparator = null;
-        try {
-            multiAttributeSeparator = CarbonContext.getThreadLocalCarbonContext().getUserRealm().
-                    getRealmConfiguration().getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR);
-        } catch (UserStoreException e) {
-            log.warn("Error while retrieving MultiAttributeSeparator from UserRealm.");
-            if (log.isDebugEnabled()) {
-                log.debug("Error while retrieving MultiAttributeSeparator from UserRealm.", e);
+        if (Boolean.parseBoolean(IdentityUtil.getProperty(MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_ENABLED))) {
+            try {
+                Attribute configAttribute = FrameworkServiceDataHolder.getInstance().getConfigurationManager()
+                        .getAttribute(MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_RESOURCE_TYPE,
+                                MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_RESOURCE_NAME,
+                                MULTI_ATTRIBUTE_SEPARATOR_TENANT_CUSTOMIZATION_ATTRIBUTE_NAME);
+
+                if (configAttribute != null && StringUtils.isNotBlank(configAttribute.getValue())) {
+                    multiAttributeSeparator = configAttribute.getValue();
+                }
+            } catch (ConfigurationManagementException e) {
+                if (!ERROR_CODE_RESOURCE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode()) &&
+                        !ERROR_CODE_ATTRIBUTE_DOES_NOT_EXISTS.getCode().equals(e.getErrorCode())) {
+                    log.warn(String.format("Error while retrieving the custom MultiAttributeSeparator " +
+                                    "for the tenant: %s. Error code: %s, Error message: %s",
+                            CarbonContext.getThreadLocalCarbonContext().getTenantDomain(), e.getErrorCode(),
+                            e.getMessage()));
+                }
+            }
+        }
+
+        if (StringUtils.isBlank(multiAttributeSeparator)) {
+            try {
+                multiAttributeSeparator = CarbonContext.getThreadLocalCarbonContext().getUserRealm().
+                        getRealmConfiguration().getUserStoreProperty(IdentityCoreConstants.MULTI_ATTRIBUTE_SEPARATOR);
+            } catch (UserStoreException e) {
+                log.warn("Error while retrieving MultiAttributeSeparator from UserRealm.");
```

Comment:
Should not this be an error message?

Reply:
This was an existing log. I followed the same pattern for the other log as well

Reply:
+1. Will change

Reply:
Fixed